### PR TITLE
Set always record events option for the RPC spans.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: core/src/main/java/io/grpc/internal/CensusTracingModule.java [hunk 1]
```diff
@@ -208,8 +208,9 @@ private static EndSpanOptions createEndSpanOptions(io.grpc.Status status) {
 
     ClientCallTracer(@Nullable Span parentSpan, String fullMethodName) {
       this.fullMethodName = checkNotNull(fullMethodName, "fullMethodName");
-      this.span =
-          censusTracer.spanBuilder(parentSpan, makeSpanName("Sent", fullMethodName)).startSpan();
+      this.span = censusTracer.spanBuilder(parentSpan, makeSpanName("Sent", fullMethodName))
+                      .setRecordEvents(true)
```
Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/internal/CensusTracingModule.java]: indent should be +4

Comment (bogdandrutu) [inline core/src/main/java/io/grpc/internal/CensusTracingModule.java]: Done.

**Comment by bogdandrutu** [comment]
PTAL.

**Comment by zhangkun83** [comment]
It should be testable in `CensusModulesTest`.

**Comment by bogdandrutu** [comment]
PTAL. Added tests.

File: core/src/test/java/io/grpc/internal/CensusModulesTest.java [hunk 2]
```diff
@@ -548,11 +550,19 @@ public void traceHeadersPropagateSpanContext() throws Exception {
...
+            return Boolean.TRUE.equals(((StartSpanOptions) argument).getRecordEvents());
+          }
+        };
+
     verify(mockTracingPropagationHandler).toBinaryValue(same(fakeClientSpanContext));
     verifyNoMoreInteractions(mockTracingPropagationHandler);
     verify(mockSpanFactory).startSpan(
         same(fakeClientParentSpan), eq("Sent.package1.service2.method3"),
-        any(StartSpanOptions.class));
+        argThat(startSpanOptionsMatcher));
```
Comment (zhangkun83) [inline core/src/test/java/io/grpc/internal/CensusModulesTest.java]: May worth changing all `any(StartSpanOptions.class)` (except the stubbing) to this

Comment (bogdandrutu) [inline core/src/test/java/io/grpc/internal/CensusModulesTest.java]: Done.

**Comment by zhangkun83** [comment]
@carl-mastrangelo, any more comments?

**Comment by bogdandrutu** [comment]
Rebased on the current head.

**Pull request merged**